### DISCUSSION TITLE
Difficulties with integrating external libraries that don't support QRL

### DISCUSSION THREAD

**Discussion by shukklin** [QwikDev/qwik-evolution#199] [External Library Ideas]
The context:
1. We want to integrate Qwik with the TanStack library. Some of the integration code was taken from this discussion https://github.com/TanStack/query/pull/5809/files.
2. TanStack does not allow accepting QRL functions, but for some attributes it works, because they accept asynchronous functions by default (for example, `queryFn` attribute accepts promise which is what QRL is).
3. But for some attributes (for example, `refetchInterval`) TanStack does not allow passing an asynchronous function, because it is not supported.
4. This leads to the fact that we cannot pass a non-asynchronous function, because Qwik cannot serialize it and we can't fully implement TanStack Query integration without hacks.

Question:
What is the best way to deal with this situation, except for creating the necessary functions on the client without serializing them in QRL (for example, inside the useVisibleTask hook which looks like too hacky)? 

It seems that it would be nice to solve it more optimally out of the box than to create various hacks.

----
Reproduction is here - https://stackblitz.com/edit/github-pqqkc67l

Steps to reproduce:
1. Run server with command - `npm start`.
2. Go to `src/routes/index.tsx` file.
3. You will see Typescript error at 13 line and that the passed function does not work correctly, because TanStack does not accept an asynchronous function as a parameter for the `refetchInterval` attribute.

**Comment by shairez** [comment]
@wmertens this might be solved by `useSerialized$` right?

**Reply by wmertens** [reply]
Yes #204
